src/pages/index.js

```diff
@@ -61,16 +107,64 @@ const addImageModal = new ModalWithForm(
 //-----------CREATE A MODALWITHIMAGE INSTANCE FOR THE PREVIEW IMAGE------------>>
 const previewModal = new ModalWithImage("#image-preview-modal");
 
+//-----------CREATE A FUNCTION THAT HANDLES DELETE CLICK------------>>
+function handleDeleteClick(card) {
+  deleteConfirmationModal.open();
+  deleteConfirmationModal.setCallback(() => {
+    api.deleteCard(card.getId()).catch((err) => console.log(err));
+    card.deleteCard();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.